File: KalininHutor.Domain.Booking/BookingRoomVariant.cs
using KalininHutor.Domain.Booking.Enums;

namespace KalininHutor.Domain.Booking;

public class BookingRoomVariant : IEntity<Guid>
{
    public Guid Id { get; protected set; }
    public Guid BookingId { get; protected set; }
    public Guid RoomVariantId { get; protected set; }
    public int RoomsCount { get; protected set; }
    public decimal Amount { get; protected set; }
    public BedTypes BedType { get; protected set; }

    protected BookingRoomVariant() { }

    public BookingRoomVariant(Guid roomVariantId, Guid bookingId, int roomsCount, decimal amount, BedTypes bedType)
    {
        if (roomVariantId == Guid.Empty)
            throw new ArgumentNullException("Не указан идентификатор варианта номера.");

        if (bookingId == Guid.Empty)
            throw new ArgumentNullException("Не указан идентификатор брони.");

        if (amount < 0)
            throw new ArgumentNullException("Цена за аренду номера не может быть отрицательной.");

        Id = Guid.NewGuid();
        RoomVariantId = roomVariantId;
        BookingId = bookingId;
        RoomsCount = roomsCount;
        Amount = amount;
        BedType = bedType;
    }
}

File: kalinin-hutor/src/models/bookingRoomVariant.ts
import { BedTypes } from ".";

// Модель чтения выбранного варианта номера
export interface BookingRoomVariant {
    // Идентификатор выбранного номера
    id?: string;
    // Идентификатор варианта номера
    roomVariantId: string;
    // Идентификатор брони
    bookingId?: string;
    //Количество бронируемых номеров
    roomsCount: number;
    //Цена за номер
    price: number;
    // Всего за номер (руб.)
    amount: number;
    //Выбранный тип кровати
    bedType: BedTypes;
}

export namespace BookingRoomVariant {
    // Очередь получения забронированных вариантов кроватей в номера
    export interface GetQuery {
        // Идентификатор выбранного номера
        id: string;
        // Идентификатор брони
        bookingId: string;
    }

    // Запрос на создание выбранного варианта номера
    export interface CreateRequest {
        // Идентификатор варианта номера
        roomVariantId: string;
        // Идентификатор брони
        bookingId?: string;
        // Всего за номер (руб.)
        amount: number;

        // Выбранные типы кроватей
        bedType: BedTypes;
    }

    // Запрос удаления выбранного варианта номера
    export interface DeleteRequest {
        // Идентификатор выбранного номера
        id: string;
    }
}